Frontend/my-vite-app/src/Components/Homepage.jsx

```diff
@@ -34,7 +34,36 @@ const homePage = ({ profile }) => {
     };
     const response = await createRequest(data);
     if (response.success) {
-      navigate("/buddy", { state: { match: response.data.matched } });
+      let timer = 0;
+      if (response.data.matched.length === 0) {
+        const searchInterval = async () => {
+          try {
+            setIsSearching(true);
+            const response = await createRequest(data);
+
+            if (response.data.matched.length > 0) {
+              setIsSearching(false);
+              navigate("/buddy", { state: { match: response.data.matched } });
+            }
+
+            if (timer <= 60000) {
+              setTimeout(searchInterval, 30000);
+              timer += 30000;
+            } else {
+              setIsSearching(false);
+              return navigate("/buddy", {
+                state: { match: response.data.matched },
+              });
+            }
+          } catch (error) {
```

Comment:
Here, you are catching the error and returning an error message. 
The issue here is:
1. You are returning a string, and it is consistent with ```return navigate("/buddy", {
                state: { match: response.data.matched },
              });```
2. You are not assigning the error message anywhere, meaning the error will be swallowed.